posts/squidpy-essentials/squidpy-essentials.qmd

```diff
@@ -0,0 +1,1071 @@
+is autocorrelated and co-occurs with cells high in _KRT19_.
+
+```{filename="Python"}
+
+sq.pl.spatial_segment(
+    adata_subset,
+    library_id="35",
+    seg_cell_id="cell_ID",
+    library_key="fov",
+    color=["COX6C", "COL1A1", "KRT19"],
+    size=60,
+    img=False,
+    figsize=(4, 4),
+    dpi = 200,
+    save = "fig-nhood-Morans-I.png"
+)
+```
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-morans"
+#| fig-width: 8
+#| fig.show: 'hold'
+#| fig-cap: "Expression for select genes showing relatively high Moran's _I_."
+
+knitr::include_graphics(c("./figures/fig-nhood-Morans-I.png"))
+```
+
+
+### Spatial co-occurance
+
+While Moran's _I_ quantifies autocorrelation of genes, spatial co-occurance with
+`gr.co_occurance` quantifies cell-level co-occurances. In example below, co-occurance
+of Leiden clusters was used (@fig-co-occurance).
+
+```{filename="Python"}
+
+sq.gr.co_occurrence(
+    adata_subset,
+    cluster_key="leiden",
+)
+
+sq.pl.co_occurrence(
+    adata_subset,
+    cluster_key="leiden",
+    clusters="1",
+    save = "fig-co-occurance1.png"
+)
+sq.pl.spatial_segment(
+    adata_subset,
+    shape="hex",
+    color="leiden",
+    library_id="35",
+    library_key="fov",
+    seg_cell_id="cell_ID",
+    img=False,
+    size=60,
+    ax = ax[1],
+    save = "fig-co-occurance2.png"
+)
+```
+
+
+
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-co-occurance"
+#| fig-width: 8
+#| fig.show: 'hold'
+#| fig-cap: "Co-occurance of Leiden clusters. Left: Leiden clusters in space for a single FOV of breast cancer tissue. Right: co-occurance probability ratio (Y) for each cluster relative to Leiden cluster 1 (orange cells in left plot) with increasing radius (X)."
+
+cowplot::ggdraw() + cowplot::draw_image("./figures/fig-co-occurance2.png", width=0.5) + 
+  cowplot::draw_image("./figures/fig-co-occurance1.png", width = 0.5, x = 0.5)
+```
+
+### Neighborhood enrichment
+
+We can take a look at neighborhood enrichment scores using `gr.nhood_enrichment` 
+and `pl.nhood_enrichment`(@fig-neighbors). 
+
+```{filename="Python"}
+#| echo: false
+
+sq.gr.nhood_enrichment(adata2, cluster_key="leiden")
+sq.pl.nhood_enrichment(
+    adata2,
+    cluster_key="leiden",
+    figsize=(5, 5),
+    dpi = 200,
+    title="Neighborhood enrichment",
+    save = "fig-nhood-enrichment.png"
+)
+
+```
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-neighbors"
+#| fig-width: 8
+#| fig.show: 'hold'
+#| fig-cap: "Neighborhood enrichment across all FOVs for each pairwise combination of Leiden clusters."
+
+knitr::include_graphics(c("./figures/fig-nhood-enrichment.png"))
+```
+
+
+<!-- # Saving annData -->
+
+<!-- ```{filename="Python"} -->
+<!-- adata2.write("./adata2.h5ad") -->
+
+<!-- ``` -->
+
+# Conclusion
+
+In this post I showed how pivot CosMx SMI data that were exported from AtoMx SIP into a format that can be read with the open-sourced python package squidpy. 
+
+
+# Appendix {.appendix}
```

Comment:
This is a quarto thing, unfortunately. Items in the appendix are flagged with a special `.appendix` tag. I suppose I could just make it a regular section since, in this particular case, the section is indeed at the end (i.e., not in the middle of the text).

Reply:
removed `.appendix` tag. 3fae1263af0a8842dd563a81c7b5d5aadba72303